Tests/Marketplace/search_and_uninstall_pack.py

```diff
@@ -377,6 +471,12 @@ def clean_machine(options: argparse.Namespace, cloud_machine: str) -> bool:
             success = uninstall_all_packs(client, cloud_machine, non_removable_packs) and \
                 wait_for_uninstallation_to_complete(client, non_removable_packs)
     success &= sync_marketplace(client=client)
+    success &= handle_delete_datasets_by_testdata(
+        base_url=base_url,
+        api_key=api_key,
+        auth_id=xdr_auth_id,
+        test_data_filenames=options.modeling_rules_to_test_files,
```

Comment:
why not:
```suggestion
        datasets=get_datasets_to_delete(options.modeling_rules_to_test_files),
```
get_datasets_to_delete is basically `get_dataset_and_packs_from_collected_modeling_rules_to_test`, note that the name is a bit confusing (also the arg name, why "test_data_filenames" if it's modeling rules?)